src/main/scala/mesosphere/marathon/MarathonModule.scala

```diff
@@ -101,6 +110,17 @@ class MarathonModule(conf: MarathonConf, http: HttpConf, zk: ZooKeeperClient)
     }
   }
 
+  @Provides
+  @Singleton
+  def provideLeadershipCallbacks(
```

Comment:
At the time of writing this store was not born - So good catch! 

Please note: if you disable the cache, all direct stores are not leadership aware.